### PR TITLE
Proxy logging

### PR DESCRIPTION
**Pull request by dlespiau**
This PR adds some tracing facilities to the proxy to help with debugging. It should have done way earlier, lesson learned.

Example with `docker run busybox ls` and maximum verbosity:

```
$ sudo ./cc-proxy -v 2
I1125 16:29:49.384830   32015 proxy.go:315] proxy started
I1125 16:29:55.250793   32015 proxy.go:59] [client #1] client connected
I1125 16:29:55.447234   32015 proxy.go:69] [client #1] hello(containerId=b97346dee717057b000e0f635c974d5fb841c18089108f7db64093b8da530f46,ctlSerial=/var/run/cc-oci-runtime/b97346dee717057b000e0f635c974d5fb841c18089108f7db64093b8da530f46/ga-ctl.sock,ioSerial=/var/run/cc-oci-runtime/b97346dee717057b000e0f635c974d5fb841c18089108f7db64093b8da530f46/ga-tty.sock
I1125 16:29:55.583992   32015 proxy.go:69] [client #1] cmd=startpod, data={"hostname":"b97346dee717057b000e0f635c974d5fb841c18089108f7db64093b8da530f46","containers":[],"shareDir":"rootfs","interfaces":[{"device":"enp0s8","newDeviceName":"eth0","ipAddresses":[{"ipAddress":"172.17.0.2","netMask":"255.255.0.0"}]}]})
I1125 16:29:55.596470   32015 proxy.go:69] [client #1] allocateIo(nStreams=2)
I1125 16:29:55.596581   32015 proxy.go:69] [client #1] -> 2 streams allocated, ioBase=1
I1125 16:29:55.616531   32015 proxy.go:59] [client #2] client connected
I1125 16:29:55.616627   32015 proxy.go:69] [client #2] attach(containerId=b97346dee717057b000e0f635c974d5fb841c18089108f7db64093b8da530f46)
I1125 16:29:55.616871   32015 proxy.go:69] [client #2] cmd=newcontainer, data={"id":"b97346dee717057b000e0f635c974d5fb841c18089108f7db64093b8da530f46","rootfs":"","image":"","restartPolicy":"never","initialize":false,"process":{"terminal":false,"stdio":1,"stderr":2,"workdir":"/","args":["ls"],"envs":[{"value":"/root","env":"HOME"},{"value":"/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin","env":"PATH"},{"value":"b97346dee717","env":"HOSTNAME"}]}})
I1125 16:29:55.625569   32015 proxy.go:59] [client #2] connection closed
I1125 16:29:55.669652   32015 vm.go:99] [vm b97346de io] <- writing 59 bytes to client #1
I1125 16:29:55.669700   32015 vm.go:106] 
00000000  00 00 00 00 00 00 00 01  00 00 00 3b 62 69 6e 0a  |...........;bin.|
00000010  64 65 76 0a 65 74 63 0a  68 6f 6d 65 0a 6c 69 62  |dev.etc.home.lib|
00000020  0a 70 72 6f 63 0a 72 6f  6f 74 0a 73 79 73 0a 74  |.proc.root.sys.t|
00000030  6d 70 0a 75 73 72 0a 76  61 72 0a                 |mp.usr.var.|
I1125 16:29:55.669724   32015 vm.go:99] [vm b97346de io] <- writing 12 bytes to client #1
I1125 16:29:55.669732   32015 vm.go:106] 
00000000  00 00 00 00 00 00 00 01  00 00 00 0c              |............|
I1125 16:29:55.669742   32015 vm.go:99] [vm b97346de io] <- writing 13 bytes to client #1
I1125 16:29:55.669747   32015 vm.go:106] 
00000000  00 00 00 00 00 00 00 01  00 00 00 0d 00           |.............|
I1125 16:29:55.669894   32015 proxy.go:59] [client #1] connection closed
I1125 16:29:55.746686   32015 proxy.go:59] [client #3] client connected
I1125 16:29:55.746758   32015 proxy.go:59] [client #3] bye()
I1125 16:29:55.783097   32015 proxy.go:59] [client #3] connection closed
```

### CODE REVIEW

**Comment by jodh-intel** [comment]
Pullapprove has caught a commit with a missing Sign-off.

**Comment by dlespiau** [comment]
It was a fixup commit (git commit --fixup) left around. FIxed now.

Fixes: https://github.com/01org/cc-oci-runtime/issues/345

File: proxy/README.md [hunk 1]
```diff
@@ -112,3 +112,23 @@ handled by systemd and can be viewed with:
...
+`glog` can be configured through command line parameters, try the `-h` option
+for more details. Contrary to the `glog` defaults, `cc-proxy` defaults to
+writing log messages to stderr .
+
+```
+$ sudo ./cc-proxy -v 2
+```
+
+Additionally, the `CC_PROXY_LOG_LEVEL` environment variable can be used to set
+the log level. The command like parameter `-v` takes precedence over the
```
Comment (jodh-intel) [inline proxy/README.md]: How many log levels are available?

File: proxy/README.md [hunk 2]
```diff
@@ -112,3 +112,23 @@ handled by systemd and can be viewed with:
...
+environment variable.
+
+```
+$ sudo CC_PROXY_LOG_LEVEL=1 ./cc-proxy
```
Comment (jodh-intel) [inline proxy/README.md]: To make this easier to control for those running the proxy as a service, how about adding something like `EnvironmentFile=` to the service file?

Comment (dlespiau) [inline proxy/README.md]: I'd rather not have it enabled by default though

**Comment by dlespiau** [comment]
Update PR:

- Added a note about the number of log levels
- Fixed typo in README

**Comment by sameo** [comment]
LGTM

**Comment by jcvenegas** [comment]
lgtm

**Comment by chavafg** [comment]
qa-failed

**Comment by chavafg** [comment]
Tests started run before rebase, so launching again

**Comment by chavafg** [comment]
qa-passed

**Comment by dlespiau** [comment]
as, hum, always, pullapprove needs lgtms after a rebase. @sameo, @jcvenegas mind giving your lgtms again?

This is starting to be a bit annoying.

**Comment by gorozco1** [comment]
👍

**Comment by jcvenegas** [comment]
lgtm

**Comment by chavafg** [comment]
qa-passed

**Pull request merged**